device/tt_umd.hpp

```diff
@@ -0,0 +1,57 @@
+/*
+ * SPDX-FileCopyrightText: (c) 2023 Tenstorrent Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
```

Comment:
Can you explain the purpose for this file?
Is the intended proposal for this to be the new API which metal would use, outside the class?
I don't get how would this fit into larger picture